service/snap/snap.go

```diff
@@ -0,0 +1,422 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package snap is a minimal service.Service implementation modelled on service/upstart.
+package snap
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"runtime"
+	"strings"
+
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	"github.com/juju/utils/featureflag"
+	"github.com/juju/utils/set"
+	"github.com/juju/utils/shell"
+
+	"github.com/juju/juju/feature"
+	"github.com/juju/juju/service/common"
+)
+
+const (
+	Command           = "snap"
+	CustomConfigPath  = "/var/snap/juju-db/common/custom.config"
+	DefaultConfigPath = "/var/snap/juju-db/common/default.config"
+	DefaultDataPath   = "/var/snap/juju-db/common"
```

Comment:
this is what needs to be the dbDir (via that helper function mentioned earlier) if snap is used